.github/workflows/ci.yaml

```diff
@@ -0,0 +1,354 @@
+name: CI
+
+on:
+  push:
+    branches: [main]
+  merge_group:
+  pull_request:
+    branches:
+      - "main"
+      - "v[0-9]+.[0-9]+.[0-9]+"
+      - "v[0-9]+.[0-9]+"
+  workflow_dispatch:
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-20.04 
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v4
+
+    - name: Verify
+      uses: golangci/golangci-lint-action@v3
+      with:
+        version: v1.51.2
+        args: --timeout=15m0s --verbose
+  
+  unit:
+    runs-on: ubuntu-20.04
+    name: Unit and Integration tests
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v4
+
+    - name: Setup
+      uses: ./.github/workflows/composite/setup
+
+    - name: Verify manifests
+      run: |
+        make manifests
+        git diff --exit-code
+
+    - name: Verify bin dir
+      run: |
+        BIN_FILE="metallb-operator.yaml" make bin
+        git diff --exit-code bin/
+
+    - name: Unit and Integration Tests
+      run: |
+        export KUBECONFIG=${HOME}/.kube/config
+        METALLB_BGP_TYPE=native make test
+        METALLB_BGP_TYPE=frr DEPLOY_KUBE_RBAC_PROXIES=true make test
+
+  e2e:
+    runs-on: ubuntu-20.04
+    env:
+      built_image: "metallb-operator:ci" # Arbitrary name
+    needs:
+      - lint
+      - unit
+    defaults:
+      run:
+        shell: bash
+        working-directory: metallboperator
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+
+    - name: Setup
+      uses: ./.github/workflows/composite/setup
+
+    - name: Build image
+      run: |
+        IMG=${built_image} make docker-build
+
+    - name: Create K8s Kind Cluster
+      run: |
+        ./hack/kind-cluster-without-registry.sh
+        kind load docker-image ${built_image}
+
+    - name: Deploy Metal LB Operator
+      run: |
+        IMG=${built_image} KUSTOMIZE_DEPLOY_DIR="config/kind-ci/" make deploy
+
+    - name: E2E Tests
+      run: |
+        export KUBECONFIG=${HOME}/.kube/config
+        make test-validation
+        make test-e2e
+
+    - name: Collect Logs
```

Comment:
did you check/see this works on failure? I wonder since the `if: ${{ failure() }}` moved to another file and I'm not sure it would run it here without it

Reply:
Let's remove the `if: ${{ failure() }}` from the collectlogs composite workflow and just do the check everyplace before we call it, like we do in metallb.

Reply:
Yes, I checked and saw that it works on failure. @oribon